pkg/resource/alloy-secret/reconciler.go

```diff
@@ -29,81 +23,12 @@ type Reconciler struct {
 
 // ReconcileCreate ensures Alloy secret is created with the right credentials
 func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Interface) (ctrl.Result, error) {
-	logger := log.FromContext(ctx)
-	logger.Info("alloy-secret - create")
-
-	if lc.GetLoggingAgent() != common.LoggingAgentAlloy {
-		logger.Info(fmt.Sprintf("alloy-secret - logging agent is not %s, skipping", common.LoggingAgentAlloy))
-		result, err := r.ReconcileDelete(ctx, lc)
-		if err != nil {
-			return result, errors.WithStack(err)
-		}
-		return result, nil
-	}
-
-	// Check existence of Alloy app
-	var currentApp appv1.App
-	appMeta := common.ObservabilityBundleAppMeta(lc)
-	err := r.Client.Get(ctx, types.NamespacedName{Name: lc.AppConfigName(common.AlloyLogAgentAppName), Namespace: appMeta.GetNamespace()}, &currentApp)
-	if err != nil {
-		if apimachineryerrors.IsNotFound(err) {
-			logger.Info(fmt.Sprintf("alloy-secret - %s app not found, requeuing", lc.AppConfigName(common.AlloyLogAgentAppName)))
-			// If the app is not found we should requeue and try again later (5 minutes is the app platform default reconciliation time)
-			return ctrl.Result{RequeueAfter: time.Duration(5 * time.Minute)}, nil
-		}
-		return ctrl.Result{}, errors.WithStack(err)
-	}
-
-	// Retrieve secret containing credentials
-	var loggingCredentialsSecret v1.Secret
-	err = r.Client.Get(ctx, types.NamespacedName{Name: loggingcredentials.LoggingCredentialsSecretMeta(lc).Name, Namespace: loggingcredentials.LoggingCredentialsSecretMeta(lc).Namespace},
-		&loggingCredentialsSecret)
-	if err != nil {
-		return ctrl.Result{}, errors.WithStack(err)
-	}
-
-	// Retrieve Loki ingress name
-	lokiURL, err := common.ReadProxyIngressURL(ctx, lc, r.Client)
-	if err != nil {
-		return ctrl.Result{}, errors.WithStack(err)
-	}
-
-	// Get desired secret
-	desiredAlloySecret, err := GenerateAlloyLoggingSecret(lc, &loggingCredentialsSecret, lokiURL)
+	result, err := r.ReconcileDelete(ctx, lc)
```

Comment:
Do you want to delete always?

Reply:
Oh yes you do

Reply:
Yes for now this should always be deleted. In the next release we can remove this resource entirely, I'll add a TODO about this.